rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -192,6 +197,14 @@ def put(self):
             source_deployment=params[SOURCE_DEPLOYMENT],
             target_deployment=params[TARGET_DEPLOYMENT],
             created_at=now)
+        if hasattr(params[SOURCE_DEPLOYMENT], '_storage_id') and \
+                hasattr(params[TARGET_DEPLOYMENT], '_storage_id'):
+            source_id = str(params[SOURCE_DEPLOYMENT]._storage_id)
+            target_id = str(params[TARGET_DEPLOYMENT]._storage_id)
+            get_resource_manager().assert_no_cyclic_dependencies(source_id,
```

Comment:
In case you're not passing the functions to utils, I think it would be better to call the resource manager once and save it to a variable.